src/FakeItEasy/Creation/CastleDynamicProxy/CastleDynamicProxyGenerator.cs

```diff
@@ -67,14 +89,11 @@ public ProxyGeneratorResult GenerateProxy(Type typeOfProxy, IEnumerable<Type> ad
                 return new ProxyGeneratorResult(DynamicProxyResources.ProxyIsSealedTypeMessage.FormatInvariant(typeOfProxy));
             }
 
+            Guard.AgainstNull(additionalInterfacesToImplement, "additionalInterfacesToImplement");
             GuardAgainstConstructorArgumentsForInterfaceType(typeOfProxy, argumentsForConstructor);
+            Guard.AgainstNull(fakeCallProcessorProvider, "fakeCallProcessorProvider");

```

Comment:
It looked weird to me too, but I felt better when I noticed that we were short-circuiting above. I guess pushing the guards into `CreateProxyGeneratorResult` will help with this as well.